src/components/ExcalidrawView.tsx

```diff
@@ -13,65 +12,172 @@ import {
   ExcalidrawImperativeAPI,
 } from '@excalidraw/excalidraw/types/types';
 
-import { DEBOUNCE_VALUE } from '../config/settings';
-import getInitialData from './InitialData';
+import {
+  APP_DATA_TYPES,
+  ExcalidrawElementsAppData,
+  ExcalidrawStateAppData,
+} from '../config/appDataTypes';
+import {
+  DEFAULT_EXCALIDRAW_ELEMENTS_APP_DATA,
+  DEFAULT_EXCALIDRAW_STATE_APP_DATA,
+} from '../config/constants';
+import {
+  DEBOUNCE_COMPARE_ELEMENTS,
+  DEBOUNCE_SAVE_ELEMENTS,
+  DEBOUNCE_SAVE_STATE,
+} from '../config/settings';
+import { reconcileElements } from '../utils/reconciliation';
 import { useAppDataContext } from './context/AppDataContext';
 
-type Props = {
-  appData: AppData;
-};
-
-const ExcalidrawView: FC<Props> = ({ appData }) => {
+const ExcalidrawView: FC = () => {
   const excalidrawRef = useRef<ExcalidrawImperativeAPI>(null);
   const viewModeEnabled = false;
   const zenModeEnabled = false;
   const gridModeEnabled = false;
   const theme = 'light';
-  const { patchAppData } = useAppDataContext();
-  // eslint-disable-next-line react/destructuring-assignment
-  const { id, data } = appData;
-  const iData = getInitialData(
-    data.elements as readonly ExcalidrawElement[],
-    // data.state as AppState,
-  );
+  const date = new Date();
+  const name = `drawing${date.toISOString()}`;
+
+  const [localElements, setLocalElements] =
+    useState<readonly ExcalidrawElement[]>();
+  const [idElements, setIdElements] = useState<string>('');
+  const [idState, setIdState] = useState<string>('');
+  const [appState, setAppState] = useState<AppState>();
+
+  const { appDataArray, patchAppData, postAppData } = useAppDataContext();
+
+  useEffect(() => {
+    const elementsAppData: ExcalidrawElementsAppData = (appDataArray.find(
+      ({ type }) => type === APP_DATA_TYPES.EXCALIDRAW_ELEMENTS,
+    ) as ExcalidrawElementsAppData) ?? {
+      type: APP_DATA_TYPES.EXCALIDRAW_ELEMENTS,
+      id: '',
+      data: { elements: [] },
+    };
+
+    const stateAppData: ExcalidrawStateAppData = (appDataArray.find(
+      ({ type }) => type === APP_DATA_TYPES.EXCALIDRAW_STATE,
+    ) as ExcalidrawStateAppData) ?? {
+      type: APP_DATA_TYPES.EXCALIDRAW_STATE,
+      id: '',
+      data: { appState: {} },
+    };
+    const { id, data: newData } = elementsAppData;
+    setIdElements(id);
+    const { elements: newElements } = newData;
+    if (excalidrawRef.current && typeof localElements !== 'undefined') {
```

Comment:
Is the type assertion necessary in this context ? Doesn't a "simple" `localElements` assertion work ? It may not haha ...